blocktx/peer_handler.go

```diff
@@ -93,32 +94,38 @@ func init() {
 }
 
 func NewPeerHandler(logger utils.Logger, storeI store.Interface, blockCh chan *blocktx_api.Block) p2p.PeerHandlerI {
-	s := &PeerHandler{
-		store:    storeI,
-		blockCh:  blockCh,
-		logger:   logger,
-		workerCh: make(chan utils.Pair[*chainhash.Hash, p2p.PeerI], 100),
-		announcedCache: expiringmap.New[chainhash.Hash, []p2p.PeerI](10 * time.Minute).WithEvictionFunction(func(hash chainhash.Hash, peers []p2p.PeerI) bool {
-			msg := wire.NewMsgGetData()
+	evictionFunc := func(hash chainhash.Hash, peers []p2p.PeerI) bool {
+		msg := wire.NewMsgGetData()
 
-			if err := msg.AddInvVect(wire.NewInvVect(wire.InvTypeBlock, &hash)); err != nil {
-				logger.Errorf("EvictionFunc: could not create InvVect: %v", err)
-				return false
-			}
+		if err := msg.AddInvVect(wire.NewInvVect(wire.InvTypeBlock, &hash)); err != nil {
+			logger.Errorf("EvictionFunc: could not create InvVect: %v", err)
+			return false
+		}
+		// Select a random peer to send the request to
+		n, err := rand.Int(rand.Reader, big.NewInt(int64(len(peers))))
```

Comment:
What is the rationale behind changing `peer := peers[rand.Intn(len(peers))]` to this?

Reply:
it was recommendation from sonarcube gosec check. I can revert it to standard math rand or we can disable this check.